src/Tizen.Sensor/Tizen.Sensor/SensorRecorder.cs

```diff
@@ -0,0 +1,504 @@
+﻿/*
+ * Copyright (c) 2019 Samsung Electronics Co., Ltd All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the License);
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an AS IS BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Tizen.Sensor
+{
+    /// <summary>
+    /// The SensorRecorder is used for retrieving the recorded data from a particular sensor.
+    /// </summary>
+    /// <since_tizen> 6 </since_tizen>
+    public class SensorRecorder : IDisposable
```

Comment:
SensorRecorder should be static because it can record only one sensor at one time.
or use singleton pattern

Reply:
replacing it with singleton pattern would be better it seems.

Reply:
@pvschakradhar Could you explain work flow of SensorRecoder?
I can't understand how to work it and how to use it 
I need information of relation of Sensor and SensorRecorder

Reply:
SensorRecorder is separate class, which is used for fetching the data stored in Data Base. There is no direct relation between Sensor and sensor recorder. Data Base differentiates the information as per sensor type.
So SensorRecorder can be made Static class or singleton class.

Reply:
How to provide a state of recording?
The recording state should separated by each sensor instance.

for example, we should provide a way to check that specific sensor is now recording state

Reply:
I analysed in depth and found that data is being received in callback function with the sensor type parameter. So with this type parameter, sensor data can be separated. Kindly suggest your opinion on this. Is it better to use separate instance per sensor or static/singleton class for all sensors.

But I found sensor data(struct ctx_sensor_rec_data_s)
is coming as a void pointer, Need to handle receiving the complete data from C API to C# API.
Here is the structure of a data in C Layer:
struct ctx_sensor_rec_data_s {
		std::vector<std::string> keys;
		std::shared_ptr<Tuple> tuple;
	};